mad-dog/src/helpers.js

```diff
@@ -79,23 +89,30 @@ const addAndUpgradeUsers = async (
     await waitForIndexing()
   })
 
-  // TODO: should check if existing users are already creators. if so, dont upgrade
   await logOps('Upgrade to creator', async () => {
     try {
       await executeAll(async (libs, i) => {
+        // Check if existing users are already creators. If so, don't upgrade
+        const existingUser = await getUser({ executeOne, walletIndex: i })
+
         // Autoselect replica set from valid nodes on-chain
-        const selectedCNodes = await executeOne(i, libsWrapper =>
-          autoSelectCreatorNodes(libsWrapper, numCreatorNodes)
-        )
-        const endpointString = makeCreatorNodeEndpointString(selectedCNodes)
-        logger.info(`Upgrading creator wallet index ${i} with ${endpointString} endpoints`)
-        // Upgrade to creator with replica set
-        await executeOne(i, l => upgradeToCreator(l, endpointString))
-        logger.info(`Finished upgrading creator wallet index ${i}`)
+        if (!existingUser || !existingUser.creator_node_endpoint) {
+          const selectedCNodes = await executeOne(i, libsWrapper =>
+            autoSelectCreatorNodes(libsWrapper, numCreatorNodes)
+          )
+          const endpointString = makeCreatorNodeEndpointString(selectedCNodes)
+          logger.info(`Upgrading creator wallet index ${i} with ${endpointString} endpoints`)
+          // Upgrade to creator with replica set
+          await executeOne(i, l => upgradeToCreator(l, endpointString))
+          logger.info(`Finished upgrading creator wallet index ${i}`)
+        } else {
+          logger.info(`User ${existingUser.user_id} is already a creator. Skipping upgrade...`)
+        }
       })
     } catch (e) {
       logger.error('GOT ERR UPGRADING USER TO CREATOR')
       logger.error(e.message)
+      throw e
```

Comment:
so this is what caused the false positive test? cause this never threw?

Reply:
yup! if you look at the logs, it prints this error message, but then never exits out of the test because it just logs the error.

Reply:
gonna add an explicit check for error